wemake_python_styleguide/visitors/ast/functions.py

```diff
@@ -186,17 +192,40 @@ def _check_argument_default_values(self, node: AnyFunctionDef) -> None:
             if not isinstance(real_arg, self._allowed_default_value_types):
                 self.add_violation(ComplexDefaultValueViolation(node))
 
+    def _check_generator(self, node: AnyFunctionDef) -> None:
```

Comment:
Let's move the generator detection logic to `logic/`, it might be reused soon.
And call it as `is_generator(node: AnyFunctionDef) -> bool` function.